clusterloader2/pkg/test/simpletestexecutor.go

```diff
@@ -28,20 +34,80 @@ func createSimpleTestExecutor() TestExecutor {
 
 // ExecuteTest executes test based on provided configuration.
 func (ste *simpleTestExecutor) ExecuteTest(ctx Context, conf *api.Config) error {
+	defer cleanupResources(ctx)
+	err := ctx.GetFramework().CreateAutomanagedNamespaces(int(conf.AutomanagedNamespaces))
+	if err != nil {
+		return fmt.Errorf("automanaged namespaces creation failed: %v", err)
+	}
+
+	for i := range conf.Steps {
+		if err := ste.ExecuteStep(ctx, &conf.Steps[i]); err != nil {
+			return err
+		}
+	}
 	return nil
 }
 
 // ExecuteStep executes single test step based on provided step configuration.
 func (ste *simpleTestExecutor) ExecuteStep(ctx Context, step *api.Step) error {
-	return nil
+	var err error
+	if len(step.Measurements) > 0 {
+		// TODO: handle measurments
+	} else {
+		var wg wait.Group
+		var lock sync.Mutex
+		for i := range step.Phases {
+			phase := &step.Phases[i]
+			wg.Start(func() {
+				if phaseErr := ste.ExecutePhase(ctx, phase); phaseErr != nil {
+					lock.Lock()
+					defer lock.Unlock()
+					if err == nil {
+						err = phaseErr
```

Comment:
Any reason why you're capturing only the first error and not all?

Reply:
All error will be logged, so no information will be missed.
In case of critical error the first one to occur should be the original error.

Reply:
Could you change this to an errList and return that instead for completeness?
That'll be useful to generate an error report at the end of the test. For simplicity, you can merge all errors into a single one at the top-level (i.e in `ExecuteTest()`) for now.

Reply:
After thinking more about this, I decided to make every `Execute...` function return `[]error`.
Every object operation can fail, so when `n` replicas are created, there could be `n` errors return. Thus we need `[]error` being propagated from the bottom to the top.